### PR TITLE
Fixed incorrect use of walrus operator

### PR DESCRIPTION
**Pull request by lasradorohan**
Incorrect use of the walrus operator caused `argnums` to be assigned a boolean instead of the value of rhs value.

**Context:**
Parameter `argnums` of `catalyst.grad()` and `catalyst.value_and_grad()` was getting ignored. The gradients returned were of the first parameter regardless of `argnums`.
In incorrect use of the walrus operator `:=` was found in the code.

**Description of the Change:**
Brackets were added to ensure correct assignment of value.

**Benefits:**
`argnums` now works as expected.

**Possible Drawbacks:**

**Related GitHub Issues:**
[#1477](https://github.com/PennyLaneAI/catalyst/issues/1477)
[sc-82697]

### CODE REVIEW

**Comment by Qottmann** [comment]
I dont think this is actually a bug. Also note that pylint complains when you do these brackets around the walrus operator.

I personally also prefer it with the brackets for better readability though, so wouldnt mind the change. But note that there are several occurences of this pattern in the codebase, so it wouldnt be just changing this line here.

**Comment by dime10** [comment]
Hi @lasradorohan, thank you so much for uncovering and even providing a fix for it 🚀 🚀

**Review by dime10 — COMMENTED** [review]
I've approved the CI to run on your PR to see if anything breaks (it will check formatting and linting). In order to merge this PR, we'll require the following:
- Normally all bug fixes require one (or more) new tests that would have failed before and now pass. This is to prevent us from regressing into the bug again in the future. If you would like to add a test case (maybe a slimmed down version of how you originally discovered it) you can add a test [in this file](https://github.com/PennyLaneAI/catalyst/blob/main/frontend/test/pytest/test_gradient.py). Otherwise, let me know and we can add a test to your PR.
- Update the changelog saying what was fixed with this PR [in this file](https://github.com/PennyLaneAI/catalyst/blob/e50fd94616843f2b506e6d825e8da04503310b8a/doc/releases/changelog-dev.md?plain=1#L15), and most importantly, don't forget to add your name to the contributor list at the bottom of the file :)

File: frontend/test/pytest/test_gradient.py [hunk 1]
```diff
@@ -1780,5 +1780,35 @@ def fn(x):
...
+    """Tests https://github.com/PennyLaneAI/catalyst/issues/1477"""
+    @qml.qnode(device=qml.device("lightning.qubit", wires=4), interface="jax")
+    def circuit(inputs, weights):
+        qml.AngleEmbedding(features=inputs, wires=range(4), rotation="X")
+        for i in range(1, 4):
+            qml.CRX(weights[i - 1], wires=[i, 0])
+        return qml.expval(qml.PauliZ(wires=0))
+
+    if transform_qjit:
+        circuit = qjit(circuit)
```
Comment (dime10) [inline frontend/test/pytest/test_gradient.py:1795]: I don't think we need to test without qjit, since we would be testing PennyLane in that case (which has its own test suite):
```suggestion
def test_grad_argnums(argnums):
    """Tests https://github.com/PennyLaneAI/catalyst/issues/1477"""
    @qjit
    @qml.qnode(device=qml.device("lightning.qubit", wires=4), interface="jax")
    def circuit(inputs, weights):
        qml.AngleEmbedding(features=inputs, wires=range(4), rotation="X")
        for i in range(1, 4):
            qml.CRX(weights[i - 1], wires=[i, 0])
        return qml.expval(qml.PauliZ(wires=0))
```

File: frontend/test/pytest/test_gradient.py [hunk 2]
```diff
@@ -1780,5 +1780,35 @@ def fn(x):
...
+    def compare_structure_and_value(o1, o2):
+        return tree_structure(o1) == tree_structure(o2) and \
+            tree_all(tree_map(jnp.allclose, o1, o2))
+
+    result = grad(circuit, argnums=argnums)(weights, inputs)
+    expected = jax.grad(circuit, argnums=argnums)(weights, inputs)
+    assert compare_structure_and_value(result, expected)
+
+    _, result = value_and_grad(circuit, argnums=argnums)(weights, inputs)
+    _, expected = jax.value_and_grad(circuit, argnums=argnums)(weights, inputs)
```
Comment (dime10) [inline frontend/test/pytest/test_gradient.py:1809]: Ideally we might want to compare against a computation not involving Catalyst at all, which can be achieved like this (applying `jax.grad` function is a sort of Catalyst/JAX hybrid):
```suggestion
    expected = jax.grad(circuit.original_function, argnums=argnums)(weights, inputs)
    assert compare_structure_and_value(result, expected)

    _, result = value_and_grad(circuit, argnums=argnums)(weights, inputs)
    _, expected = jax.value_and_grad(circuit.original_function, argnums=argnums)(weights, inputs)
```

**Review by dime10 — APPROVED** [review]
Nice work, happy to approve 💯 (looks like you just need to run `black frontend` to satisfy the CI)
